ugs-platform/ugs-platform-visualizer/src/main/java/com/willwinder/ugs/nbm/visualizer/renderables/GcodeModel.java

```diff
@@ -287,62 +288,73 @@ private boolean generateObject()
      * Convert the gcodeLineList into vertex and color arrays.
      */
     private void updateVertexBuffers() {
-        if (this.isDrawable) {
-            Color color;
-            int vertIndex = 0;
-            int colorIndex = 0;
-            byte[] c = new byte[3];
-            for(LineSegment ls : gcodeLineList) {
-                // Find the lines color.
-                if (ls.isArc()) {
-                    color = arcColor;
-                } else if (ls.isFastTraverse()) {
-                    color = rapidColor;
-                } else if (ls.isZMovement()) {
-                    color = plungeColor;
-                } else {
-                    color = linearColor;
-                }
+        // Prevent concurrent updates which will queue events
+        if(isUpdatingVertexBuffers) {
```

Comment:
There's technically a race condition here, I think this function needs to be synchronized, and set the isUpdatingVertexBiffers flag outside the invokeLater.